core/common/src/main/java/alluxio/util/network/NettyUtils.java

```diff
@@ -83,9 +93,16 @@ public static EventLoopGroup createEventLoop(ChannelType type, int numThreads,
    * Returns the correct {@link SocketChannel} class based on {@link ChannelType}.
    *
    * @param type Selector for which form of low-level IO we should use
-   * @return SocketChannel matching the ChannelType
+   * @param isDomainSocket whether this is to connect to a domain socket server
+   * @return Channel matching the requirements
    */
-  public static Class<? extends SocketChannel> getClientChannelClass(ChannelType type) {
+  public static Class<? extends Channel> getClientChannelClass(ChannelType type,
+      boolean isDomainSocket) {
+    if (isDomainSocket) {
+      Preconditions.checkState(type == ChannelType.EPOLL,
+          "Domain socket is used while the channel type is not epoll.");
```

Comment:
Would "Domain sockets are only supported with EPOLL channel type" be more clear?